### PR TITLE
Special some handle history client generation

### PR DESCRIPTION
**Pull request by wxing1292**
**What changed?**
* Special handle history client VerifyChildExecutionCompletionRecorded API generation, due to this request contains
  * ParentExecution, which contains WorkflowId
  * ChildExecution, which also contains WorkflowId
  * pathToField function will use the first element which contains WorkflowId, for shard redirection
  * pathToField function behavior is ambiguous for shard redirection, e.g. using workflow ID from parent or child execution?
* Special handle history client RecordChildExecutionCompleted, due to same reason
  * Rename some variables making the meaning clear 

**Why?**
Make some logic more resilient to bugs

**How did you test it?**
Make bins

**Potential risks**
N/A

**Is hotfix candidate?**
No, since we are lucky

### CODE REVIEW

File: cmd/tools/rpcwrappers/main.go [hunk 1]
```diff
@@ -93,9 +93,10 @@ var (
 		"retryableClient.history.StreamWorkflowReplicationMessages": true,
 
 		// these are non-standard implementations. do not generate.
-		"client.history.DescribeHistoryHost":    true,
-		"client.history.GetReplicationMessages": true,
-		"client.history.GetReplicationStatus":   true,
+		"client.history.DescribeHistoryHost":                    true,
+		"client.history.GetReplicationMessages":                 true,
+		"client.history.GetReplicationStatus":                   true,
+		"client.history.VerifyChildExecutionCompletionRecorded": true,
```
Comment (yux0) [inline cmd/tools/rpcwrappers/main.go:99]: nit: do you also need to add StreamWorkflowReplicationMessages?

Comment (wxing1292) [reply cmd/tools/rpcwrappers/main.go:99]: the override is on line # 91 https://github.com/temporalio/temporal/pull/4708/files#diff-224d80d92f0d36d3692f634944dc3462611a85efe56db8b8546f8662521308f2R91

**Review by yycptt — APPROVED** [review]
Do we need to pull `RecordChildExecutionCompleted` out as well?

**Review by yux0 — APPROVED** [review]

**Comment by dnr** [comment]
>   * pathToField function will use the first element which contains WorkflowId, for shard redirection
>   * pathToField function behavior is ambiguous for shard redirection, e.g. using workflow ID from parent or child execution?

I can change this so that it's an error if it finds two WorkflowId fields. That would prevent potential bugs like this. Sound like a good idea?

**Comment by wxing1292** [comment]
> > * pathToField function will use the first element which contains WorkflowId, for shard redirection
> > * pathToField function behavior is ambiguous for shard redirection, e.g. using workflow ID from parent or child execution?
> 
> I can change this so that it's an error if it finds two WorkflowId fields. That would prevent potential bugs like this. Sound like a good idea?

definitely a good idea

**Pull request merged**